src/mnelab/utils/channelstats_calculations.py

```diff
@@ -0,0 +1,39 @@
+# © MNELAB developers
+#
+# License: BSD (3-clause)
+
+from collections import defaultdict
+
+import numpy as np
+from mne import channel_type
+from mne.defaults import _handle_default
+
+
+def calculate_channel_stats(raw):
+    nchan = raw.info["nchan"]
+    cols = defaultdict(list)
+    cols["name"] = raw.ch_names
+
+    for i in range(nchan):
+        ch = raw.info["chs"][i]
+        data = raw[i][0]
+        cols["type"].append(channel_type(raw.info, i))
+        cols["unit"].append(ch.get("unit", ""))
+        cols["min"].append(np.min(data))
+        cols["Q1"].append(np.percentile(data, 25))
+        cols["mean"].append(np.mean(data))
+        cols["median"].append(np.median(data))
+        cols["Q3"].append(np.percentile(data, 75))
+        cols["max"].append(np.max(data))
```

Comment:
It takes quite some time until the dialog window pops up, presumably because the calculation here uses a loop over channels. Can you use vectorized NumPy calls instead, e.g., if `data` is the entire 2D NumPy data array, you can do `np.mean(data, axis=1)` to get the mean for all channels simultaneously.